network/src/peers/peers.rs

```diff
@@ -356,79 +369,88 @@ impl<S: Storage + Send + Sync + 'static> Node<S> {
+            0
+        };
+
+        for remote_address in self.connected_peers() {
+            self.peer_book.sending_ping(remote_address);
+
+            self.outbound
+                .send_request(Message::new(
+                    Direction::Outbound(remote_address),
+                    Payload::Ping(current_block_height),
+                ))
+                .await;
+        }
+    }
+
     /// TODO (howardwu): Implement manual serializers and deserializers to prevent forward breakage
     ///  when the PeerBook or PeerInfo struct fields change.
     ///
     /// Stores the current peer book to the given storage object.
     ///
-    /// This function checks that this node is not connected to itself,
-    /// and proceeds to serialize the peer book into a byte vector for storage.
+    /// This function serializes the peer book into a byte vector for storage.
     ///
     #[inline]
-    fn save_peer_book_to_storage(&self) -> Result<(), NetworkError> {
+    pub(crate) fn save_peer_book_to_storage(&self) -> Result<(), NetworkError> {
         // Serialize the peer book.
         let serialized_peer_book = bincode::serialize(&SerializedPeerBook::from(&self.peer_book))?;
 
-        // TODO: the peer book should be stored outside of consensus
-        if let Some(ref consensus) = self.consensus() {
+        // TODO: the peer book should be stored outside of sync
+        if let Some(ref sync) = self.sync() {
             // Save the serialized peer book to storage.
-            consensus.storage().save_peer_book_to_storage(serialized_peer_book)?;
+            sync.storage().save_peer_book_to_storage(serialized_peer_book)?;
         }
 
         Ok(())
     }
 
-    /// TODO (howardwu): Add logic to remove the active channels
-    ///  and handshakes of the peer from this struct.
-    /// Sets the given remote address in the peer book as disconnected from this node server.
+    ///
+    /// Removes the given remote address channel and sets the peer in the peer book
+    /// as disconnected from this node server.
     ///
     #[inline]
     pub(crate) fn disconnect_from_peer(&self, remote_address: SocketAddr) -> Result<(), NetworkError> {
-        debug!("Disconnecting from {} (if not disconnected yet)", remote_address);
-
-        if let Some(ref consensus) = self.consensus() {
+        if let Some(ref sync) = self.sync() {
             if self.peer_book.is_syncing_blocks(remote_address) {
-                consensus.finished_syncing_blocks();
+                sync.finished_syncing_blocks();
             }
         }
 
-        self.outbound.channels.write().remove(&remote_address);
-
-        let result = self.peer_book.set_disconnected(remote_address);
-        debug!("Set {} as disconnected", remote_address);
-        result
-        // TODO (howardwu): Attempt to blindly send disconnect message to peer.
+        // Remove the peer from the channel.
+        if self.outbound.channels.write().remove(&remote_address).is_some() {
```

Comment:
limiting the disconnect to this is fine _unless_ it's disconnecting from a peer that is not `connected` but still `connecting`, otherwise there will be lingering `connecting` entries

Reply:
Discussed offline, updated